samples/star-wars-api/HttpHandlers.fs

```diff
@@ -8,277 +8,350 @@ open System.Text.Json.Serialization
+            let isMultipartRequest (req : HttpRequest) =
+                not (System.String.IsNullOrEmpty(req.ContentType)) && req.ContentType.Contains("multipart/form-data")
+
+            if requestIsEmpty then
+                let! result = executeIntrospectionQuery ValueNone
+                let response = result |> toResponse
+                return Results.Ok response
+    
+            elif isMultipartRequest request then
+
+                /// Look for a header specifying the boundary string used in a multi-part form submission and return the string if it exists.
+                let getMultipartRequestBoundary (req : HttpRequest) =
+                    req.Headers.GetCommaSeparatedValues("Content-Type")
+                    |> Seq.map (fun v -> v.TrimStart())
+                    |> Seq.tryFind (fun v -> v.Contains("boundary"))
+                    |> Option.map (fun v -> v.Remove(0, v.IndexOf('=') + 1))
+                    |> Option.map (fun v -> v.Trim('"'))
+
+                match getMultipartRequestBoundary request with
+                | Some boundary ->
+
+                    let copyBodyToMemory (req : HttpRequest) =
+                        let ms = new MemoryStream(4096)
+                        req.Body.CopyTo(ms)
+                        ms.Position <- 0L
+                        ms
+
+                    return! task {
+                        use ms = copyBodyToMemory request
+                        let reader = MultipartReader(boundary, ms)
+                        let! multiRequest = reader |> MultipartRequest.read ctx.RequestAborted
+                        let results =
+                            multiRequest.Operations
+                            |> List.map (fun op ->
+
+                                let resultTask = executeOperation op
+                                let result = resultTask.Result  // TODO: Am I doing this right?
+
+                                let addRequestType (requestType : string) (response : GQLExecutionResult) =
+                                    let mapper (content : GQLResponseContent) =
+                                        let dataMapper (data : Output) : Output =
+                                            let data = data |> Seq.map (|KeyValue|) |> Map.ofSeq
+                                            upcast data.Add("requestType", requestType)
+                                        match content with
+                                        | GQLResponseContent.Direct (data, errors) -> Direct (dataMapper data, errors)
+                                        | GQLResponseContent.Deferred (data, errors, deferred) -> Deferred (dataMapper data, errors, deferred)
+                                        | _ -> content
+                                    { Content = mapper response.Content; DocumentId = response.DocumentId; Metadata = response.Metadata }
+
+                                result |> addRequestType "Multipart"
```

Comment:
John just extended the resulting JSON with a custom field.
I do not see any benefits of doing that.
But if we really want we can copy `GQLResponse` and add an additional field